Assets/Src/Scripts/Input/WebCamSource.cs

```diff
@@ -0,0 +1,158 @@
+      _webCam = null;
+    }
+  }
+  //DropdownChange event
+  public void CameraDropdownChange(int value) {
+    //Debug.Log(_devicesName.ToArray()[value]);
+    if(_devicesName.ToArray()[value] != null) {
+      DeviceName = _devicesName.ToArray()[value];
+    }
+  }
+  //DropdownChange event
+  public void ResolutionDropdownChange(int value) {
+    string _resolution = _resolutionOpt.ToArray()[value];
+    if(_resolution != null) {
+      switch(_resolution) {
+        case "1280X720":
+          _width = 1280;
+          _height = 720;
+          break;
+        case "1600X900":
+          _width = 1600;
+          _height = 900;
+          break;
+        case "1920X1080":
+          _width = 1920;
+          _height = 1080;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(_width);
+    }
+  }
+  //DropdownChange event
+  public void FpsDropdownChange(int value) {
+    string _fps = _fpsOpt.ToArray()[value];
+    if(_fps != null) {
+      switch(_fps) {
+        case "60Hz":
+          Fps = 60;
+          break;
+        case "30Hz":
+          Fps = 30;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(Fps);
+    }
+  }
+  //DropdownChange event
+  public void FlipDropdownChange(int value) {
+    string _flip = _flipOpt.ToArray()[value];
+    if(_flip != null) {
+      switch(_flip) {
+        case "默认":
+          Flip = false;
+          break;
+        case "水平翻转":
+          Flip = true;
+          break;
+        default:
+          break;
+      }
+      //Debug.Log(Flip);
+    }
+  }
+  //Webcam and params init
+  IEnumerator Start() {
+    yield return Application.RequestUserAuthorization (UserAuthorization.WebCam);
+    if(Application.HasUserAuthorization(UserAuthorization.WebCam)) {
+      WebCamDevice[] _devices = WebCamTexture.devices;
+      CameraDropdown.ClearOptions();
+      for(int i = 0; i< _devices.Length; i++) {
+        _devicesName.Add(_devices[i].name);
+      }
+      CameraDropdown.AddOptions(_devicesName);
+      CameraDropdown.value = 0;
+      CameraDropdown.captionText.text = CameraDropdown.options[0].text;
+      CameraDropdown.onValueChanged.AddListener(CameraDropdownChange);
+    }
+
+    ResolutionDropdown.ClearOptions();
+    ResolutionDropdown.AddOptions(_resolutionOpt);
+    ResolutionDropdown.value = 0;
+    ResolutionDropdown.captionText.text = ResolutionDropdown.options[0].text;
+    ResolutionDropdown.onValueChanged.AddListener(ResolutionDropdownChange);
+
+    FpsDropdown.ClearOptions();
+    FpsDropdown.AddOptions(_fpsOpt);
+    FpsDropdown.value = 0;
+    FpsDropdown.captionText.text = FpsDropdown.options[0].text;
+    FpsDropdown.onValueChanged.AddListener(FpsDropdownChange);
+
+    FlipDropdown.ClearOptions();
+    FlipDropdown.AddOptions(_flipOpt);
+    FlipDropdown.value = 0;
+    FlipDropdown.captionText.text = FlipDropdown.options[0].text;
+    FlipDropdown.onValueChanged.AddListener(FlipDropdownChange);
```

Comment:
Code refactor: line 125-128, line 131-135, line 137-141, line 143-147 - those 5 code pieces share the same logic. Please extract the common logic to a method and convert those code pieces to 5 method calls.